scipy/optimize/tests/_cython_examples/meson.build

```diff
@@ -0,0 +1,18 @@
+project('random-build-examples', 'c', 'cpp', 'cython')
+
+py_mod = import('python')
+py3 = py_mod.find_installation(pure: false)
+
+cc = meson.get_compiler('c')
+cy = meson.get_compiler('cython')
+
+# Keep synced with pyproject.toml
```

Comment:
This should also remain working with older Cython versions, since we're exporting Cython API for other projects to use - and those other projects are not required to upgrade Cython in sync with our version bumps. Hence, I'd just drop this comment; we can leave the lower version bound at 3.0.8 for a long time.